python/cudf/cudf/core/dataframe.py

```diff
@@ -3748,20 +3772,41 @@ def any(self, bool_only=None, **kwargs):
             )
         return self._apply_support_method("any", **kwargs)
 
-    def _apply_support_method(self, method, *args, **kwargs):
-        result = [
-            getattr(self[col], method)(*args, **kwargs)
-            for col in self._cols.keys()
-        ]
-        if isinstance(result[0], Series):
-            support_result = result
-            result = DataFrame()
-            for idx, col in enumerate(self._cols.keys()):
-                result[col] = support_result[idx]
-        else:
-            result = Series(result)
-            result = result.set_index(self._cols.keys())
-        return result
+    def _apply_support_method(self, method, axis=0, *args, **kwargs):
+        assert axis in (None, 0, 1)
+
+        if axis in (None, 0):
+            result = [
+                getattr(self[col], method)(*args, **kwargs)
+                for col in self._cols.keys()
+            ]
+
+            if isinstance(result[0], Series):
+                support_result = result
+                result = DataFrame()
+                for idx, col in enumerate(self._cols.keys()):
+                    result[col] = support_result[idx]
+            else:
+                result = Series(result)
+                result = result.set_index(self._cols.keys())
+            return result
+
+        elif axis == 1:
+            # for dask metadata compatibility
+            kwargs.pop("skipna", None)
```

Comment:
Before doing this can we assert that the value of `skipna` is something where this won't cause incorrect results? I.E. if `skipna` is false we throw an exception, if `skipna` is true we do this (I think that's the correct behavior, please correct me if I'm wrong).

Reply:
Agreed. Very good point.

Reply:
Adding the `skipna` check after popping the value. I don't love how this relies on skipna being based by keyword only, but I think it's fine since this is an internal method and it can be controlled reasonably well.